homepage/tests/courses_methods_tests.py

```diff
@@ -0,0 +1,208 @@
+    except ValidationError:
+        flag = True
+
+    assert flag
+
+
+@pytest.mark.parametrize("invalid_courses", [
+    (1, "Linear Algebra 1", True, 4, None, 3.5, None, 5),
+    #                                       ^- avg_load without avg_rating
+    (1, "Linear Algebra 1", True, 4, None, None, 4, 10),
+    #                                             ^- avg_rating without avg_load
+    (1, "Linear Algebra 1", True, 4, None, None, None, 2),
+    #                                                   ^- no ratings but raters num is > 0
+    (1, "Linear Algebra 1", True, 4, None, 3.5, 4),
+    #                                            ^- rating reviews without rater numbers
+    (1, "Linear Algebra 1", True, 4, None, None, None, 0, 13),
+    #                                                      ^- reviewes without rating reviews
+    ])
+@pytest.mark.django_db
+def test_invalid_course_creation(invalid_courses):
+    course = Course(*invalid_courses)
+    course.save()
+
+    assert Course.objects.filter(pk=course.course_id).count() == 0
+
+
+# test the print_details function
+@pytest.mark.django_db
+def test_course_print_details(capsys):
+    course = (1, "Linear Algebra 1", True, 4, None, 3.500, 4.000, 13, 7)
+    Course(*course).save()
+
+    msg = (
+        "------------------------------------------------------------\n"
+        "Course indentifier: 1   \nName: Linear Algebra 1\nMandatory? yes\n"
+        "Credit Points: 4\nSyllabi: N/A\n"
+        "Average Rating: 4.000 \tAverage Load: 3.500\t"
+        "13 Raters\nNumber Of Reviews: 7\n"
+        )
+    # make sure that message comes out as expected
+    Course.objects.get(pk=1).print_details()
+    assert capsys.readouterr().out == msg
+
+
+# test the get_details function
+@pytest.mark.django_db
+def test_course_get_details():
+    course = (1, "Linear Algebra 1", True, 4, "please contribute", 3.500, 4.000, 13, 7)
+    Course(*course).save()
+    # make sure details are saved accurately
+    assert Course.objects.get(pk=course[0]).get_details() == course
+
+
+# creating multiple new courses
+@pytest.mark.django_db
+def test_create_multiple_courses():
+    details = [(1, "Linear Algebra 1", True, 4, "please contribute", 3.5, 4, 13, 7),
+               (2, "Project Management", False, 3, "please contribute", 1.5, 3.5, 15, 3)]
+
+    for args in details:
+        Course(*args).save()
+    # make sure that multiple courses are saved accurately
+    courses = [Course.objects.get(pk=arg[0]).get_details() for arg in details]
+    assert courses == details
+
+
+@pytest.fixture
+def courses():
+    courses_list = [
+         Course(1, "Course1", True, 3, None),
+         Course(2, "Course2", False, 2, None, 3.5, 2, 13, 5),
+         Course(3, "Course3", False, 1, None, 3.5, 2, 13, 0)
+         ]
+    for course in courses_list:
+        course.save()
+
+    return courses_list
+
+
+# updating courses after review
+@pytest.mark.django_db
+def test_upd_course_with_previous_data(courses):
```

Comment:
if you can make it clear and it is not too many extra characters I prefer it like that
```suggestion
def test_update_course_with_previous_data(courses):
```